### PR TITLE
Fallback to application/json encoding

### PR DESCRIPTION
**Pull request by ekroon**
This is the quick fix for #80

The proper fix is a bit harder as there is no direct support for accept header parsing in the Axum ecosystem.

### CODE REVIEW

File: columnq/src/io/mod.rs [hunk 1]
```diff
@@ -21,7 +21,9 @@ impl TryFrom<Option<&uriparse::Scheme<'_>>> for BlobStoreType {
     fn try_from(scheme: Option<&uriparse::Scheme<'_>>) -> Result<Self, Self::Error> {
```
Comment (ekroon) [inline columnq/src/io/mod.rs:21]: This file changed because of running `cargo fmt`

File: roapi-http/Cargo.toml [hunk 2]
```diff
@@ -47,7 +47,6 @@ simd = ["columnq/simd"]
 snmalloc = ["snmalloc-rs"]
 
 [dev-dependencies]
-actix-rt = "*"
```
Comment (houqp) [inline roapi-http/Cargo.toml:50]: oops, good catch :+1:

File: roapi-http/src/api/mod.rs [hunk 3]
```diff
@@ -50,18 +50,12 @@ pub fn bytes_to_json_resp(bytes: Vec<u8>) -> Response<Body> {
...
-            encoding::ContentType::try_from(hdr_value.as_bytes()).map_err(|_| ApiErrResp {
-                code: http::StatusCode::BAD_REQUEST,
-                error: "unsupported_content_type".to_string(),
-                message: format!("{:?} is not a supported response content type", hdr_value),
-            })
-        }
+        None => encoding::ContentType::Json,
+        Some(hdr_value) => encoding::ContentType::try_from(hdr_value.as_bytes())
+            .or::<()>(Ok(ContentType::Json))
+            .unwrap(),
```
Comment (houqp) [inline roapi-http/src/api/mod.rs:58]: How about simplify this with unwrap_or?

```suggestion
        Some(hdr_value) => encoding::ContentType::try_from(hdr_value.as_bytes()).unwrap_or(ContentType::Json),
```

**Comment by ekroon** [comment]
@houqp thanks for the quick review, I accepted your feedback. We seem to be hitting an error in the build, which I also got numerous times on my local machine, so I created rust-lang/rust-clippy#7970 

The suggested workaround in the error is `cargo clean -p roapi-http`. Not running clippy is not a solution, because `cargo build` also fails on the same error. I checked with rustc `1.57.0-beta.3` and it seems to be fixed then.

So options are:

- Wait for 1.57 release in December
- Clean build (`cargo clean -p roapi-http`, `cargo clean -p columnq`, `cargo clean -p columnq-cli`
- Temporarily switch to 1.57 beta, or to an older version of rust (which I didn't check)

**Comment by houqp** [comment]
@ekroon let's go with the manual cargo clean path for now. I think we might be able to also fix this by just adding a new prefix to the cache key to force a cache purge.

**Comment by ekroon** [comment]
I think having a cache is actually very useful, locally I will be switching to beta for development for the time being. Otherwise you end up having to clean a lot, breaking the development flow somewhat.

**Comment by ekroon** [comment]
@houqp tower-rs/tower-http#112 keeps being force pushed which breaks the build, I updated Cargo.lock again

**Review by houqp — APPROVED** [review]

**Comment by houqp** [comment]
Thanks @ekroon !

**Pull request merged**